docs/sdk/src/reference_docs/tokens_in_substrate.rs

```diff
@@ -0,0 +1,97 @@
+// This file is part of polkadot-sdk.
+//
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Tokens in Substrate
+//!
+//! This reference doc serves as a high-level overview of the token-related logic in Substrate and
+//! how to properly apply it to your use case.
+//!
+//! On completion of reading this doc, you should have a good understanding of:
+//! - The distinction between token traits and trait implementations in Substrate, and why this
+//!   distinction is helpful
+//! - Token-related traits avaliable in Substrate
+//! - Token-related trait implementations in Substrate
+//! - How to choose the right trait or trait implementation for your use case
+//! - Where to go next
+//!
+//! ## Traits and Trait Implementations
+//!
+//! Broadly speaking, token logic in Substrate can be divided into two categories: traits and
+//! trait implementations.
+//!
+//! **Traits** define common interfaces that types of tokens should implement. For example, the
+//! [`frame_support::traits::fungible::Inspect`] trait specifies an interface for *inspecting*
+//! token state such as the total issuance of the token, the balance of individual accounts, etc.
+//!
+//! **Trait implementations** are concrete implementations of these traits. For example, one of the
+//! many traits [`pallet_balances`] implements is [`frame_support::traits::fungible::Inspect`].
+//!
+//! The distinction between traits and trait implementations is helpful because it allows pallets
+//! and other logic to be generic over their dependencies, avoiding cumbersome pallet tight
+//! coupling.
+//!
+//! To illustrate this with an example let's consider [`pallet_preimage`]. This pallet takes a
```

Comment:
Similarly, overlaps with `frame_pallet_coupling.rs` 🙈 ofc it is fine to re-elaborate here, but I suggest read what is already present there, and see how the two docs can be made more cohesive.

Reply:
@liamaharon there is an interesting meta observation here. I think great docs are written with fewer number of authors, because it is easier to make sure everyone is reading everyone else's content and builds on top of it. In other words, a small group of authors can have a "shared vision" with less effort.

So going forward, I will make sure to report any ref doc that I am working on somewhere where everyone can see, and we should all do the same.